### DISCUSSION TITLE
maintain state within modal forms

### DISCUSSION THREAD

**Discussion by grahampcharles** [mantinedev/mantine#1921] [Feature requests]
Currently, controlled state is not maintained within modals; that is, a code fragment like this will fail:

```
modals.openModal({
            title: "State is not being reflected within the modal",
            children: (
                <TextInput
                  value={value}
                  onChange={(event) => setValue(event.currentTarget.value)}
                  data-autofocus
                />
            )
          });
```

[See this codesandbox for a full working example of this issue.](https://codesandbox.io/s/fervent-leavitt-ijqx21?file=/src/SampleComponent.tsx)

In plain-vanilla React, this same issue exists and is fixed by appending the stateful value to the `useModal` call:

```
useModal(() => (<ReactModal>...</ReactModal>), [value])
```

There does not seem to be a way to do that within a Mantine modal, at least that I can see. Any thoughts?

**Comment by rtivital** [comment]
I doubt that we will be able to support this feature – it is not possible with current implementation. You can create a wrapper component that will contain state and use it instead.
